multimodal/src/autogluon/multimodal/data/process_multimodal.py

```diff
@@ -0,0 +1,227 @@
+import ast
+import logging
+import os
+import warnings
+from copy import deepcopy
+from lib2to3.pgen2.token import OP
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from nptyping import NDArray
+from omegaconf import DictConfig
+from torch import nn
+from transformers import AutoConfig, AutoTokenizer, BertTokenizer, CLIPTokenizer, ElectraTokenizer
+
+from ..constants import (
+    AUTOMM,
+    CHOICES_IDS,
+    LABEL,
+    NER,
+    NER_ANNOTATION,
+    TEXT,
+    TEXT_SEGMENT_IDS,
+    TEXT_TOKEN_IDS,
+    TEXT_VALID_LENGTH,
+    TOKEN_WORD_MAPPING,
+    WORD_OFFSETS,
+)
+from .collator import Pad, Stack
+from .utils import process_ner_annotations, tokenize_ner_text
+
+logger = logging.getLogger(AUTOMM)
+
+# Disable tokenizer parallelism
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+ALL_TOKENIZERS = {
+    "bert": BertTokenizer,
+    "clip": CLIPTokenizer,
+    "electra": ElectraTokenizer,
+    "hf_auto": AutoTokenizer,
+}
+
+
+class MultiModalProcessor:
```

Comment:
Some functions/attributes are same as those in TextProcessor. Could we inherit from TextProcessor and reuse them?

Reply:
Maybe also inherit from LabelProcessor.

Reply:
It might be ok to inherit from them for the moment but I am not sure if there will be any conflicts or issues when it is getting used more often.

Reply:
If there would be conflicts, copying code from other processors into the multimodal seems not be able to resolve them. We will need to deal with the possible conflicts no matter doing inheritance or copying.

Reply:
resolved [here](https://github.com/awslabs/autogluon/pull/2220/commits/b66cdb2aa7df0ecce0003a941a965c00a8012ce2)